sw/device/lib/dif/dif_aes_unittest.cc

```diff
@@ -374,28 +376,94 @@ TEST_F(DataTest, DataOut) {
                                            {AES_STATUS_OUTPUT_VALID_BIT, true},
                                        });
 
-  for (uint32_t i = 0; i < ARRAYSIZE(data_.data); ++i) {
-    ptrdiff_t offset = AES_DATA_OUT_0_REG_OFFSET + (i * sizeof(uint32_t));
-    EXPECT_READ32(offset, data_.data[i]);
-  }
+  setExpectedReadMultreg(AES_DATA_OUT_0_REG_OFFSET, data_.data,
+                         ARRAYSIZE(data_.data));
 
   dif_aes_data_t out;
   EXPECT_DIF_OK(dif_aes_read_output(&aes_, &out));
 
   EXPECT_THAT(out.data, ElementsAreArray(data_.data));
 }
 
+class DataProcessTest : public AesTestInitialized {
+ protected:
+  const dif_aes_data_t dataIn[3] = {
+      {.data = {0xA55AA55A, 0xA55AA55A, 0xA55AA55A, 0xA55AA55A}},
+      {.data = {0xA55AA55A, 0xA55AA55A, 0xA55AA55A, 0xA55AA55A}},
+      {.data = {0xA55AA55A, 0xA55AA55A, 0xA55AA55A, 0xA55AA55A}},
+  };
+
+  const dif_aes_data_t dataOut[3] = {
+      {.data = {0xB44BB44B, 0xB44BB44B, 0xB44BB44B, 0xB44BB44B}},
+      {.data = {0xB44BB44B, 0xB44BB44B, 0xB44BB44B, 0xB44BB44B}},
+      {.data = {0xB44BB44B, 0xB44BB44B, 0xB44BB44B, 0xB44BB44B}},
+  };
+};
+
+TEST_F(DataProcessTest, ThreeBlocksSuccess) {
+  {
+    EXPECT_READ32(AES_STATUS_REG_OFFSET, {{AES_STATUS_INPUT_READY_BIT, true},
+                                          {AES_STATUS_OUTPUT_VALID_BIT, true}});
+
+    setExpectedWriteMultreg(AES_DATA_IN_0_REG_OFFSET, dataIn[0].data,
+                            ARRAYSIZE(dataIn[0].data));
+    EXPECT_READ32(AES_STATUS_REG_OFFSET, {{AES_STATUS_INPUT_READY_BIT, true}});
+
+    setExpectedWriteMultreg(AES_DATA_IN_0_REG_OFFSET, dataIn[1].data,
+                            ARRAYSIZE(dataIn[1].data));
+    EXPECT_READ32(AES_STATUS_REG_OFFSET, {{AES_STATUS_OUTPUT_VALID_BIT, true}});
+    setExpectedReadMultreg(AES_DATA_OUT_0_REG_OFFSET, dataOut[0].data,
+                           ARRAYSIZE(dataOut[0].data));
+
+    setExpectedWriteMultreg(AES_DATA_IN_0_REG_OFFSET, dataIn[2].data,
+                            ARRAYSIZE(dataIn[2].data));
+    EXPECT_READ32(AES_STATUS_REG_OFFSET, {{AES_STATUS_OUTPUT_VALID_BIT, true}});
+    setExpectedReadMultreg(AES_DATA_OUT_0_REG_OFFSET, dataOut[1].data,
+                           ARRAYSIZE(dataOut[1].data));
+
+    EXPECT_READ32(AES_STATUS_REG_OFFSET, {{AES_STATUS_OUTPUT_VALID_BIT, true}});
+    setExpectedReadMultreg(AES_DATA_OUT_0_REG_OFFSET, dataOut[2].data,
+                           ARRAYSIZE(dataOut[2].data));
+  }
+
+  dif_aes_data_t out[3];
+  EXPECT_DIF_OK(dif_aes_process_data(&aes_, dataIn, out, ARRAYSIZE(dataIn)));
+
+  for (int i = 0; i < ARRAYSIZE(dataIn); i++) {
+    EXPECT_THAT(out[i].data, ElementsAreArray(dataOut[i].data));
+  }
+}
+
+TEST_F(DataProcessTest, OneBlockSuccess) {
+  {
```

Comment:
What are these braces for?

Reply:
To make clear where is the test preparation

Reply:
We don't do this in any of our other tests, and it suggests to readers that the braces are significant for the purposes of e.g. lifetimes (such as for destructors).

Reply:
I think this got dropped. Would you mind dropping the braces? They have no semantic language-level meaning and will confuse readers (and we don't do this in other code).

Reply:
This got resolved again without it getting applied. Is this a problem with Github's UI?

Reply:
I'm not sure what happed, I think I have fixed it in the wrong commit (Next). But Know it is fixed.